From 1dd2e614b20962173bd05519479e6271ae9cb4f0 Mon Sep 17 00:00:00 2001
From: Miroslav Rezanina <mrezanin@redhat.com>
Date: Wed, 31 Jul 2013 08:04:01 +0200
Subject: [PATCH 042/212] pc: Drop RHEL-6 compat_props {apic, kvm-apic}.vapic

Message-id: <1375201922-6794-8-git-send-email-armbru@redhat.com>
Patchwork-id: 52816
O-Subject: [RHEL-7 PATCH v3 07/15] pc: Drop RHEL-6 compat_props {apic, kvm-apic}.vapic
Bugzilla: 983991
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>

RHEL-6's kvm_arch_create() calls kvm_tpr_opt_setup() #ifdef
KVM_EXIT_TPR_ACCESS, which is defined in RHEL-6.  Adds migration
section "kvm-tpr-opt" when KVM is enabled.  Comes from qemu-kvm,
initial commit 885e90b.

It went into upstream qemu v1.1.0, but in different form (commit
e5ad936 "kvmvapic: Introduce TPR access optimization for Windows
guests").  Here, migration section "kvm-tpr-opt" belongs to qdev
"kvmvapic", which is created when the property kvm-apic.vapic
(inherited from apic_common) is on.  Commit 9bea6a2 disables it for
older machine types, with suitable compat_props.

Drop them from the RHEL-6 machine types.

Signed-off-by: Markus Armbruster <armbru@redhat.com>
---
 hw/i386/pc_piix.c |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index 00170cf..3c9c4a5 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -1064,10 +1064,6 @@ static QEMUMachine pc_machine_rhel700 = {
         .property = "class",\
         .value    = stringify(PCI_CLASS_MEMORY_RAM),\
     },{\
-        .driver   = "apic",\
-        .property = "vapic",\
-        .value    = "off",\
-    },{\
         .driver   = "virtio-blk-pci",\
         .property = "event_idx",\
         .value    = "off",\
-- 
1.7.1

